syntax = "proto3";
package version;




import "model/artifact/version.proto";

/*
GetVersionList请求
*/
message GetVersionListRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 pageSize = 2 ;
    /*
    排序字段
    */
    string order = 3 ;
    /*
    程序包Id
    */
    string packageId = 4 ;
    /*
    版本类型 1 开发, 3 测试, 7 预发布, 15 生产
    */
    int32 env_type = 5 ;
    /*
    包名称
    */
    string name = 6 ;
}

/*
GetVersionList返回
*/
message GetVersionListResponse {
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated artifact.Version list = 4 ;
}

/*
GetVersionListApi返回
*/
message GetVersionListResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    version.GetVersionListResponse data = 4 ;
}

